frontend/packages/metal3-plugin/src/components/host-detail.tsx

```diff
@@ -178,19 +190,33 @@ const BaremetalHostDetails: React.FC<BaremetalHostDetailsProps> = ({
   );
 };
 
-export const BaremetalHostDetailPage: React.FC<BaremetalHostDetailPageProps> = (props) => {
-  const machinesResource = {
-    kind: referenceForModel(MachineModel),
-    namespaced: true,
-    isList: true,
-    prop: 'machines',
-  };
-  const nodesResource = {
-    kind: NodeModel.kind,
-    namespaced: false,
-    isList: true,
-    prop: 'nodes',
-  };
+const BaremetalHostDetailPage: React.FC<BaremetalHostDetailPageProps> = (props) => {
+  const { hasNodeMaintenanceCapability } = props;
```

Comment:
IMO this should be the generally preferred way to declare typed functional components.

If we have access to ES destructuring syntax, let's destructure the `props` function argument instead of doing it in the function's body.